### PR TITLE
Add condition to show link to suggest organisation

### PR DESCRIPTION
**Pull request by eduarda-meirinhos-bjss**

### CODE REVIEW

File: src/modules/shared/pages/innovation/data-sharing-and-support/data-sharing-and-support.component.ts [hunk 1]
```diff
@@ -249,6 +254,22 @@ export class PageInnovationDataSharingAndSupportComponent extends CoreComponent
         }
       }
 
+      // Check if there are organisations to be suggested by the qualifying accessor
+      if (this.userType === UserRoleEnum.QUALIFYING_ACCESSOR) {
+        const userUnitId = this.authenticationStore.getUserContextInfo()?.organisationUnit?.id || '';
+
+        const engagingUnitsIds = results.innovationSupports
+          .filter(support => support.status === 'ENGAGING')
```
Comment (diogo-costa-bjss) [inline src/modules/shared/pages/innovation/data-sharing-and-support/data-sharing-and-support.component.ts:262]: We should use the Enum: `InnovationSupportStatusEnum`

File: src/modules/shared/pages/innovation/data-sharing-and-support/data-sharing-and-support.component.ts [hunk 2]
```diff
@@ -249,6 +254,22 @@ export class PageInnovationDataSharingAndSupportComponent extends CoreComponent
         }
       }
 
+      // Check if there are organisations to be suggested by the qualifying accessor
+      if (this.userType === UserRoleEnum.QUALIFYING_ACCESSOR) {
+        const userUnitId = this.authenticationStore.getUserContextInfo()?.organisationUnit?.id || '';
```
Comment (diogo-costa-bjss) [inline src/modules/shared/pages/innovation/data-sharing-and-support/data-sharing-and-support.component.ts:259]: I think we could use the `this.stores.authentication` from the core component, no need to inject the store through constructor.

File: src/modules/shared/pages/innovation/support/support-summary-list.component.ts [hunk 3]
```diff
@@ -143,6 +160,20 @@ export class PageInnovationSupportSummaryListComponent extends CoreComponent imp
 
         LocalStorageHelper.setObjectItem(lsCacheId, Array.from(this.lsCache));
 
+        // Check if there are organisations to be suggested by the qualifying accessor
+        if (this.isQualifyingAccessorRole) {
+          const userUnitId = this.authenticationStore.getUserContextInfo()?.organisationUnit?.id || '';
```
Comment (diogo-costa-bjss) [inline src/modules/shared/pages/innovation/support/support-summary-list.component.ts:165]: Same as in the other component

**Review by diogo-costa-bjss — CHANGES_REQUESTED** [review]

File: src/modules/shared/pages/innovation/data-sharing-and-support/data-sharing-and-support.component.ts (hunk 2 above)
Comment (eduarda-meirinhos-bjss) [reply src/modules/shared/pages/innovation/data-sharing-and-support/data-sharing-and-support.component.ts:259]: You are correct! I forgot about this. Thanks

File: src/modules/shared/pages/innovation/data-sharing-and-support/data-sharing-and-support.component.ts (hunk 1 above)
Comment (eduarda-meirinhos-bjss) [reply src/modules/shared/pages/innovation/data-sharing-and-support/data-sharing-and-support.component.ts:262]: Yes!!

File: src/modules/shared/pages/innovation/support/support-summary-list.component.ts (hunk 3 above)
Comment (eduarda-meirinhos-bjss) [reply src/modules/shared/pages/innovation/support/support-summary-list.component.ts:165]: Thanks!

**Review by diogo-costa-bjss — APPROVED** [review]

**Pull request merged**